resolwe_bio/tools/validate_fastq.py

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env python3
+"""Validate input FASTQ file(s)."""
+
+import argparse
+import os
+
+import dnaio
+
+from resolwe_runtime_utils import error, warning
+from dnaio.exceptions import FastqFormatError, FileFormatError
+
+
+SUPPORTED_EXTENSIONS = (
+    '.fastq',
+    '.fastq.gz',
+    '.fq',
+    '.fq.gz',
+)
+
+def parse_arguments():
+    """Parse command line arguments."""
+    parser = argparse.ArgumentParser(description="Validate single/paired-end FASTQ files.")
+    parser.add_argument('-fq', nargs='+', type=str, required=True, help="FASTQ file(s).")
+    parser.add_argument('-fq2', nargs='+', type=str, required=False, help="Second-in-pair FASTQ file(s).")
+    return parser.parse_args()
+
+
+def basename(path):
+    """Return the base name of pathname path."""
+    return os.path.basename(path)
+
+
+def main():
+    """Invoke when run directly as a program."""
+    args = parse_arguments()
+
+    if args.fq2:
+        input_fastq = args.fq + args.fq2
+    else:
+        input_fastq = args.fq
+
+    for fq_file in input_fastq:
+        if not os.path.isfile(fq_file):
+            print(error('Input file {} does not exist'.format(basename(fq_file))))
+            exit(1)
+        if not fq_file.lower().endswith(SUPPORTED_EXTENSIONS):
+            print(error('Supported file name extensions are {}.'.format(SUPPORTED_EXTENSIONS)))
+            exit(1)
+
+    if len(set(input_fastq)) != len(input_fastq):
+        print(error('Duplicate input file names detected.'))
+        exit(1)
+
+    if args.fq2 and len(args.fq) != len(args.fq2):
+        print(error('The number of mate-pair files in split-lane samples must match. '
+                    '{} and {} input files were given for the -fq and -fq2 inputs, '
+                    'respectively.'.format(len(args.fq), len(args.fq2))))
+        exit(1)
+
+    if args.fq2:
+        for mate1, mate2 in zip(args.fq, args.fq2):
+            try:
+                paired_reads = dnaio.open(mate1, file2=mate2, fileformat='fastq')
+                if not any(True for r1, r2 in zip(paired_reads.reader1._iter, paired_reads.reader2._iter)):
```

Comment:
Just use `if not any(list(paired_reads))`.

Reply:
I know this is ugly, but if one would use ``list(reads_reader_object)`` here, the memory consumptions would explode for the large input FASTQ files. @mzganec did you have some other simplifications in mind?

Reply:
True, indeed, But you could still use simpler syntax for iteration over `paired_reads` object:
`any(True for r1, r2 in paired_reads))`

Reply:
> @mzganec did you have some other simplifications in mind?

`if not any(paired_reads):`

Reply:
@mzganec Thanks. This works for the single-end parsing. I had to retain the ``zip()`` for mate-pair parsing, since the iterator gets out of sync for some reason in the ``else`` part.